print-bgp.c

```diff
@@ -2592,6 +2682,22 @@ bgp_capabilities_print(netdissect_options *ndo,
                tok2str(bgp_safi_values, "Unknown", GET_U_1(opt + i + 5)),
                GET_U_1(opt + i + 5));
             break;
+        case BGP_CAPCODE_BGPSEC:
+            /* Version (4 bits), Direction (1 bit), Flags (3 bits), AFI (16 bits) */
+            cap_offset = 1;
+            if (cap_len < 3) {
```

Comment:
Suggested change:
```patch
diff --git a/print-bgp.c b/print-bgp.c
index 70082ef2..765ed7a7 100644
--- a/print-bgp.c
+++ b/print-bgp.c
@@ -2685,8 +2685,9 @@ bgp_capabilities_print(netdissect_options *ndo,
         case BGP_CAPCODE_BGPSEC:
             /* Version (4 bits), Direction (1 bit), Flags (3 bits), AFI (16 bits) */
             cap_offset = 1;
-            if (cap_len < 3) {
-                ND_PRINT(" (too short, < 3)");
+            /* The capability length [...] MUST be set to 3. */
+            if (cap_len != 3) {
+                ND_PRINT(" [%u != 3]", cap_len);
                 return;
             }
 
```

Reply:
[pr_942_2a.patch.txt](https://github.com/the-tcpdump-group/tcpdump/files/9693370/pr_942_2a.patch.txt)